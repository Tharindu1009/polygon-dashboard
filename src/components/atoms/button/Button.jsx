
import './styles/button.scss';
import * as React from 'react';
import CN from "classnames";

function Button({ label, type, handleClick }) {

    const ButtonClasses = CN({
        "btn btn__default": type === "default" || "submit",
        "btn btn__dark": type === "dark",
    });

    return (
        <button data-testid="button" type={type === "submit" ? type : "button"} className={ButtonClasses}
            onClick={() => handleClick && handleClick()}> {label}
        </button>
    );
}

export default Button;

