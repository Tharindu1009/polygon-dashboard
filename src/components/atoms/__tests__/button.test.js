import { render, screen, cleanup } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import Button from "../button/Button";

afterEach(()=>{
    cleanup();
})

test('should render button component', () => {
    render(<Button type="submit" label="Log in" />);
    const btnElement = screen.getByTestId('button');
    expect(btnElement).toBeInTheDocument();
    expect(btnElement).toHaveTextContent('Log in');
})

test('matches button snapshot', () => {
    const tree = TestRenderer.create(<Button type="submit" label="Log in" />).toJSON();
    expect(tree).toMatchSnapshot();
})