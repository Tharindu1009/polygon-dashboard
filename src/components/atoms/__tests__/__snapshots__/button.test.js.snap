// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches button snapshot 1`] = `
<button
  className="btn btn__default"
  data-testid="button"
  onClick={[Function]}
  type="submit"
>
   
  Log in
</button>
`;

exports[`matches snapshot 1`] = `
<button
  className="btn btn__default"
  data-testid="button"
  onClick={[Function]}
  type="submit"
>
   
  Log in
</button>
`;
